app-games/src/main/java/com/aptoide/android/aptoidegames/appview/AppViewScreen.kt

```diff
@@ -98,9 +98,20 @@ const val appViewRoute = "app/{source}"
 fun appViewScreen() = ScreenData.withAnalytics(
   route = appViewRoute,
   screenAnalyticsName = "AppView",
-  deepLinks = listOf(navDeepLink { uriPattern = BuildConfig.DEEP_LINK_SCHEMA + appViewRoute })
+  deepLinks = listOf(
+    navDeepLink { uriPattern = BuildConfig.DEEP_LINK_SCHEMA + appViewRoute },
+    navDeepLink { uriPattern = "https://{source}.{lang}.aptoide.com/{path}" },
+    navDeepLink { uriPattern = "https://{lang}.aptoide.com/{path}?{source}" }
+  )
 ) { arguments, navigate, navigateBack ->
-  val source = arguments?.getString("source")!!
+  val path = arguments?.getString("path") ?: ""
+
+  val source = when (path) {
+    "download" -> arguments?.getString("source")!!.split("&").first()
```

Comment:
In the download case, the source is part of the original Uri query, which means the order might change and it might not be the first query parameter. One possibility could be to check each possible argument of the app meta request hierarchically.